README.adoc

```diff
@@ -148,83 +155,84 @@ Write a simple `Consumer` class that consumes the artist JSON using both JSON-P
+Write the `getJsonString` and `getTotalAlbums` methods in the `ArtistResource` class. The `getTotalAlbums`
+method will use JSON-B to return the total number of albums present in the JSON for a particular artist
+and -1 if this artist does not exist using JSON-B.  The `getJsonString` method will return the JSON as
+a string that we will use later for testing.
+
+`start/src/main/java/io/openliberty/guides/consumingrest/service/ArtistResource.java`:
 
 [source, java, indent=0]
 ----
-include::finish/src/main/java/io/openliberty/guides/consumingrest/Consumer.java[tags=consumeWithJackson]
+include::finish/src/main/java/io/openliberty/guides/consumingrest/service/ArtistResource.java[tags=getJsonString]
+include::finish/src/main/java/io/openliberty/guides/consumingrest/service/ArtistResource.java[tags=getTotalAlbums]
 ----
 
-To use Jackson's JAX-RS JSON provider, you must explicitly register it like the following line since
-automatic discovery and registration are not available in the JAX-RS client API:
+=== Processing JSON using JSON-P
 
-[source, java]
+Write the method for consuming the REST service using JSON-P in the `Consumer` class:
+
+[source, java, indent=0]
 ----
-client.register(JacksonJsonProvider.class);
+include::finish/src/main/java/io/openliberty/guides/consumingrest/Consumer.java[tags=consumeWithJsonp]
 ----
 
-To bind the JSON into an `Artist` array, use the `Artist[]` entity type in the `readEntity` call:
+Write the `collectArtists` and `collectAlbums` helpers. These methods will parse that JSON and
+collect its objects into individual POJOs.  Notice that we are able to use our custom constructors
+to create instances of `Artist` and `Album`:
 
-[source, java, indent = 0]
+[source, java, indent=0]
 ----
-Artist[] artists = response.readEntity(Artist[].class);
+include::finish/src/main/java/io/openliberty/guides/consumingrest/Consumer.java[tags=collectArtists;collectAlbums]
 ----
 
-Write the `getTotalAlbums` method in the `ArtistResource` class. This method will return the total
-number of albums present in the JSON for a particular artist and -1 if this artist does not exist.
+Write the `getTotalArtists` method in the `ArtistResource` class. This method will use the JSON-P
+approach to return the total number of artists present in the JSON.
 
 `start/src/main/java/io/openliberty/guides/consumingrest/service/ArtistResource.java`:
 
 [source, java, indent=0]
 ----
-include::finish/src/main/java/io/openliberty/guides/consumingrest/service/ArtistResource.java[tags=getTotalAlbums]
+include::finish/src/main/java/io/openliberty/guides/consumingrest/service/ArtistResource.java[tags=getTotalArtists]
 ----
 
+It is worth noting that in this project the methods that we created in the `Consumer`
+class could be written directly in the `ArtistResource` class.  However, if you are
+consuming a REST resource from a third party service it is best practice to separate
+your GET and PULL requests from your data consumption.
```

Comment:
Suggested rewrite:
"The methods that you created in the `Consumer` class can be written directly in the `ArtistResource` class. However, if you are consuming a REST resource from a third party service, you can separate your GET and PULL requests from your data consumption."